### PR TITLE
Help on focus

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?170957

Should be merged same day as https://github.com/dimagi/Vellum/pull/451

@sravfeyn 

cc @amsagoff

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js-custom/hq.legacy.js [hunk 1]
```diff
@@ -13,19 +13,26 @@ var COMMCAREHQ = (function () {
...
-                $('i', el).data(attr, opts[attr]);
+                $('a', el).data(attr, opts[attr]);
             });
             if (wrap) {
                 el.hqHelp();
             }
             return el;
         },
+        transformHelpTemplate: function ($template, wrap) {
+            'use strict';
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js-custom/hq.legacy.js]: Unnecessary directive "use strict".

File: corehq/apps/hqwebapp/static/hqwebapp/js/main.js [hunk 2]
```diff
@@ -193,19 +193,26 @@ var COMMCAREHQ = (function () {
...
-                $('i', el).data(attr, opts[attr]);
+                el.find('a').data(attr, opts[attr]);
             });
             if (wrap) {
                 el.hqHelp();
             }
             return el;
         },
+        transformHelpTemplate: function ($template, wrap) {
+            'use strict';
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/main.js]: Unnecessary directive "use strict".

File: corehq/apps/style/static/style/js/commcarehq.js [hunk 3]
```diff
@@ -46,6 +47,13 @@ COMMCAREHQ.makeHqHelp = function (opts, wrap) {
     return el;
 };
 
+COMMCAREHQ.transformHelpTemplate = function ($template, wrap) {
+    'use strict';
+    var $help = COMMCAREHQ.makeHqHelp($template.data(), wrap);
+    $help.insertAfter($template);
+    $template.remove();
+}
```
Comment (dimagimon) [inline corehq/apps/style/static/style/js/commcarehq.js]: Missing semicolon.

**Comment by sravfeyn** [comment]
Looks good :+1: from me. Does this need any kind of QA though, since there are lot of style changes?

**Comment by orangejenny** [comment]
I eventually grepped for `.popover`, `icon-question-sign`, and `hq-help` and looked at each instance (didn't change every popover, only the ones that use the question icon and aren't doing something weird like putting the icon on a button, which obviously we don't want you to have to click to get at the help). I did locally test each changed page.

Agree that it's a broad change, although the likely failure modes aren't major issues:
- Question icon appears in a strange place or not at all (because a missed icon doesn't play nicely with the updated styles)
- Question icon doesn't respond to clicking (because a missed icon was a span, or was inside a label, or something else that means it doesn't handle click events)
- JavaScript error breaks the page because makeHqHelp isn't available when it's called (this is a big deal, but it should have been caught in my testing the pages where I added makeHqHelp calls)

@christineawhite thoughts on QA?

**Comment by orangejenny** [comment]
Oh, missed the part about UAT: the Vellum changes did go through UAT, so the idea of having to click on the icon instead of hover has been vetted.

**Comment by christineawhite** [comment]
@orangejenny we can probably do some quick QA by poking around for 20-30 min.  I dont think it's high risk.  Can you make me a ticket?  Should be able to get this done today.

**Comment by orangejenny** [comment]
@christineawhite Great, thank you! Just pushed to staging & submitted a ticket.

**Pull request merged**